AElf.Sdk.CSharp.Tests/TestSdkCSharpAElfModule.cs

```diff
@@ -1,26 +1,44 @@
-﻿using AElf.Database;
+﻿using System.Threading.Tasks;
+using AElf.Common;
+using AElf.Database;
+using AElf.Kernel;
 using AElf.Kernel.Infrastructure;
-using AElf.Kernel.SmartContract;
+using AElf.Kernel.SmartContract.Application;
 using AElf.Kernel.SmartContract.Contexts;
 using AElf.Modularity;
 using AElf.TestBase;
 using Microsoft.Extensions.DependencyInjection;
+using Moq;
 using Volo.Abp.Modularity;
 
 namespace AElf.Sdk.CSharp.Tests
 {
     [DependsOn(
-        typeof(DatabaseAElfModule),
         typeof(TestBaseAElfModule),
-        typeof(SmartContractAElfModule))]
-    public class TestSdkCSharpAElfModule: AElfModule
+        typeof(TestBaseKernelAElfModule))]
+    public class TestSdkCSharpAElfModule : AElfModule
     {
         public override void ConfigureServices(ServiceConfigurationContext context)
         {
             var services = context.Services;
-            context.Services.AddKeyValueDbContext<BlockchainKeyValueDbContext>(o => o.UseInMemoryDatabase());
-            context.Services.AddKeyValueDbContext<StateKeyValueDbContext>(o => o.UseInMemoryDatabase());
+            
             services.AddTransient<IStateProviderFactory, StateProviderFactory>();
```

Comment:
do not config here

Reply:
I found another test class (ContractTest) using this IStateProviderFactory. If want to remove this config, this test project need to depends on SmartContractAElfModule, can I do it like this?

Reply:
StateProviderFactory also need a lot of other dependency, it need blockchain state manager.

if you want to mock, mock IStateProviderFactory and 
ISmartContractRunner. but you know, mock need to make it work like the interface defined

Reply:
no, I do not want to mock StateProviderFactory. 
services.AddTransient<IStateProviderFactory, StateProviderFactory>() is here just because 
ContractTest want to get StateProviderFactory from ioc.
If remove it , the test project(TestSdkCSharpAElfModule) must depends on SmartContractAElfModule, because IStateProviderFactory is injected in SmartContractAElfModule. I'm not sure it's right or not？

Reply:
It has been removed.  and depends on SmartContractAElfModule.